src/shape/shared_shape.rs

```diff
@@ -319,7 +319,7 @@ impl SharedShape {
         vertices: &[Point<Real>],
         indices: &[[u32; DIM]],
         params: &VHACDParameters,
-    ) -> Self {
+    ) -> Option<Self> {
```

Comment:
This is a bit complicated to promote to fallible, because the input may contain multiple incorrect shapes.

behavior on master is to only return the shapes that the algorithm could make, and crash when NO shapes are able to be made.

This PR improves this by not crashing. It may be interesting to have more information about why it failed (and how to fix).

Reply:
add a todo to improve error reporting